### PR TITLE
[10713] DDS API unit tests

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Merge after #1822

### CODE REVIEW

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+ * 12. Delete DDS entities.
+ * The case where the delete_datawriter fails returning RETCODE_ERROR is not checked because it only passes the error
+ * upstream.
+ */
+TEST(StatisticsDomainParticipantTests, EnableDisableStatisticsDataWriterTest)
+{
+eprosima::fastdds::dds::DataWriterQos inconsistent_qos = eprosima::fastdds::dds::DATAWRITER_QOS_DEFAULT;
+inconsistent_qos.reliability().kind = eprosima::fastdds::dds::BEST_EFFORT_RELIABILITY_QOS;
+inconsistent_qos.ownership().kind = eprosima::fastdds::dds::EXCLUSIVE_OWNERSHIP_QOS;
+
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:110]: Create standard participant here

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+#ifndef FASTDDS_STATISTICS
+    // 1. Compilation flag not set
+    eprosima::fastdds::statistics::dds::DomainParticipant statistics_participant;
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:113]: Manually cast standard participant to statistics one.

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+            create_participant(0, eprosima::fastdds::dds::PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+#ifndef FASTDDS_STATISTICS
+    eprosima::fastdds::statistics::dds::DomainParticipant * statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_EQ(statistics_participant, nullptr);
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:56]:
```suggestion
    ASSERT_EQ(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant),
            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+    participant = nullptr;
+    statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_EQ(statistics_participant, nullptr);
+
+    const_participant = nullptr;
+    const_statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(const_participant);
+    EXPECT_EQ(const_statistics_participant, nullptr);
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_ERROR);
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:78]:
```suggestion
    ASSERT_EQ(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(statistics_participant),
            eprosima::fastrtps::types::ReturnCode_t::RETCODE_ERROR);
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+#else
+    logError(STATISTICS_DOMAINPARTICIPANT_TEST, "This test is going to fail because API is not yet implemented.")
+
+    // 2. Create DomainParticipant and narrow to eprosima::fastdds::statistics::dds::DomainParticipant
+    eprosima::fastdds::dds::DomainParticipant * participant =
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            create_participant(0, eprosima::fastdds::dds::PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:135]: Move to the beginning of the test

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+            new eprosima::fastdds::statistics::Entity2LocatorTrafficPubSubType);
+    eprosima::fastdds::dds::TypeSupport count_type(
+            new eprosima::fastdds::statistics::EntityCountPubSubType);
+    eprosima::fastdds::dds::TypeSupport discovery_type(
+            new eprosima::fastdds::statistics::DiscoveryTimePubSubType);
+    eprosima::fastdds::dds::TypeSupport sample_identity_count_type(
+            new eprosima::fastdds::statistics::SampleIdentityCountPubSubType);
+    eprosima::fastdds::dds::TypeSupport physical_data_type(
+            new eprosima::fastdds::statistics::PhysicalDataPubSubType);
+
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:158]: I would check here that the types are not registered and the topics do not exist

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+            RTPS_SENT_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(RTPS_SENT_TOPIC));
+    EXPECT_EQ(rtps_traffic_type, statistics_participant->find_type(rtps_traffic_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            RTPS_LOST_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(RTPS_LOST_TOPIC));
+    // This topic uses the rtps_traffic_type that has been checked previously.
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:194]: It does no harm to check it is still present. Same for all the comments like this one.

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 8]
```diff
@@ -0,0 +1,436 @@
...
+}
+
+/*
+ * This test checks both eprosima::fastdds::statistics::dds::DomainParticipant enable_statistics_datawriter() and
+ * disable_statistics_datawriter() methods.
+ * 1. If the FASTDDS_STATISTICS compilation flag is not set, check that they return RETCODE_UNSUPPORTED.
+ * 2. Create a eprosima::fastdds::dds::DomainParticipant and narrow to the children class.
+ * 3. Enable every single statistics datawriter successfully returning RETCODE_OK.
+ * 4. Check that the corresponding topic has been created and the type has been registered.
+ * 5. Enable and already enabled statistics DataWriter and check that it returns RETCODE_OK.
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:89]:
```suggestion
 * 5. Enable an already enabled statistics DataWriter and check that it returns RETCODE_OK.
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 9]
```diff
@@ -0,0 +1,436 @@
...
+    EXPECT_EQ(sample_identity_count_type, statistics_participant->find_type(
+            sample_identity_count_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            PHYSICAL_DATA_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PHYSICAL_DATA_TOPIC));
+    EXPECT_EQ(physical_data_type, statistics_participant->find_type(physical_data_type.get_type_name()));
+
+    // 5. Enable an enabled statistics DataWriter
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:263]:
```suggestion
    // 5. Enable an already enabled statistics DataWriter
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 10]
```diff
@@ -0,0 +1,436 @@
...
+            "INVALID_TOPIC", STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription("INVALID_TOPIC"));
+
+    // 7. Disable statistics DataWriter
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(HISTORY_LATENCY_TOPIC));
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PDP_PACKETS_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:279]: As the null type support check is done several times, add a `null_type` variable and compare with it

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 11]
```diff
@@ -0,0 +1,436 @@
...
+
+    // RETCODE_BAD_PARAMETER error has precedence over RETCODE_INCONSISTENT_POLICY
+    ret = statistics_participant->enable_statistics_datawriter(
+            "INVALID_TOPIC", inconsistent_qos);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+
+    // 9. Try to delete DomainParticipant and check that it fails because statistics DataWriters are still enabled
+    EXPECT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_PRECONDITION_NOT_MET);
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:298]:
```suggestion
    EXPECT_EQ(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
            delete_participant(statistics_participant),
            eprosima::fastrtps::types::ReturnCode_t::RETCODE_PRECONDITION_NOT_MET);
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 12]
```diff
@@ -0,0 +1,436 @@
...
+
+    // 10. Disable already disabled DataWriter
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:302]:
```suggestion
    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 13]
```diff
@@ -0,0 +1,436 @@
...
+
+    // 11. Disable invalid topic name
+    ret = statistics_participant->disable_statistics_datawriter("INVALID_TOPIC");
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+
+    // 12. Remove DDS entities
+    ret = statistics_participant->disable_statistics_datawriter(NETWORK_LATENCY_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:310]: This and all the ASSERT_EQ below should be EXPECT_EQ

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 14]
```diff
@@ -0,0 +1,436 @@
...
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(NETWORK_LATENCY_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            network_latency_type.get_type_name()));
+
+    ret = statistics_participant->disable_statistics_datawriter(PUBLICATION_THROUGHPUT_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(PUBLICATION_THROUGHPUT_TOPIC));
+    // The type is being used by another topic yet
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:318]: Then check it is still present. Same for similar comments below.

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 15]
```diff
@@ -0,0 +1,436 @@
...
+    ret = statistics_participant->disable_statistics_datawriter(PHYSICAL_DATA_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(PHYSICAL_DATA_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            physical_data_type.get_type_name()));
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+#endif // FASTDDS_STATISTICS
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:417]: Delete the standard participant in both cases.

```suggestion
#endif // FASTDDS_STATISTICS

    EXPECT_EQ(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
            delete_participant(participant),
            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 4 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:78]: N/A

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 16]
```diff
@@ -0,0 +1,437 @@
...
+{
+    eprosima::fastdds::dds::DomainParticipant * participant =
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            create_participant(0, PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+#ifndef FASTDDS_STATISTICS
+    eprosima::fastdds::dds::statistics::DomainParticipant * statistics_participant =
+            eprosima::fastdds::dds::statistics::DomainParticipant::narrow(participant);
+    EXPECT_EQ(statistics_participant, nullptr);
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in e11835be266a742ca3d2aa570977bd37e720274b

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 17]
```diff
@@ -0,0 +1,437 @@
...
+
+    ASSERT_TRUE(DomainParticipantFactory::get_instance()->delete_participant(participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in e11835be266a742ca3d2aa570977bd37e720274b

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 18]
```diff
@@ -0,0 +1,436 @@
...
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_NE(statistics_participant, nullptr);
+
+    const eprosima::fastdds::dds::DomainParticipant * const_participant = participant;
+    const eprosima::fastdds::statistics::dds::DomainParticipant * const_statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(const_participant);
+    EXPECT_NE(const_statistics_participant, nullptr);
+
+    participant = nullptr;
+    statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in e11835be266a742ca3d2aa570977bd37e720274b

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:118]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 2 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 3 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in e11835be266a742ca3d2aa570977bd37e720274b

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 5 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 6 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:196]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 7 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 8 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 9 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 10 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 11 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 12 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 13 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 14 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:333]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp (hunk 15 above)
Comment (JLBuenoLopez-eProsima) [reply test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp]: Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 19]
```diff
@@ -0,0 +1,436 @@
...
+    EXPECT_NE(const_statistics_participant, nullptr);
+
+    eprosima::fastdds::dds::DomainParticipant* null_participant = nullptr;
+    statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(null_participant);
+    EXPECT_EQ(statistics_participant, nullptr);
+
+    const_participant = nullptr;
+    const_statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(const_participant);
+    EXPECT_EQ(const_statistics_participant, nullptr);
+
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:79]: These tests can always be performed (independent of conditional compilation)

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 20]
```diff
@@ -0,0 +1,436 @@
...
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+
+    ret = statistics_participant->disable_statistics_datawriter("INVALID_TOPIC");
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_UNSUPPORTED);
+#else
+    logError(STATISTICS_DOMAINPARTICIPANT_TEST, "This test is going to fail because API is not yet implemented.")
+
+    // 2. Create DomainParticipant and narrow to eprosima::fastdds::statistics::dds::DomainParticipant
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:140]:
```suggestion
    // 2. Narrow the participant to eprosima::fastdds::statistics::dds::DomainParticipant
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 21]
```diff
@@ -0,0 +1,436 @@
...
+    // Check that the types are not registered yet
+    EXPECT_NE(history_latency_type, statistics_participant->find_type(history_latency_type.get_type_name()));
+    EXPECT_NE(network_latency_type, statistics_participant->find_type(network_latency_type.get_type_name()));
+    EXPECT_NE(throughput_type, statistics_participant->find_type(throughput_type.get_type_name()));
+    EXPECT_NE(rtps_traffic_type, statistics_participant->find_type(rtps_traffic_type.get_type_name()));
+    EXPECT_NE(count_type, statistics_participant->find_type(count_type.get_type_name()));
+    EXPECT_NE(discovery_type, statistics_participant->find_type(discovery_type.get_type_name()));
+    EXPECT_NE(sample_identity_count_type, statistics_participant->find_type(
+                sample_identity_count_type.get_type_name()));
+    EXPECT_NE(physical_data_type, statistics_participant->find_type(physical_data_type.get_type_name()));
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:176]:
```suggestion
    EXPECT_EQ(null_type, statistics_participant->find_type(history_latency_type.get_type_name()));
    EXPECT_EQ(null_type, statistics_participant->find_type(network_latency_type.get_type_name()));
    EXPECT_EQ(null_type, statistics_participant->find_type(throughput_type.get_type_name()));
    EXPECT_EQ(null_type, statistics_participant->find_type(rtps_traffic_type.get_type_name()));
    EXPECT_EQ(null_type, statistics_participant->find_type(count_type.get_type_name()));
    EXPECT_EQ(null_type, statistics_participant->find_type(discovery_type.get_type_name()));
    EXPECT_EQ(null_type, statistics_participant->find_type(
                sample_identity_count_type.get_type_name()));
    EXPECT_EQ(null_type, statistics_participant->find_type(physical_data_type.get_type_name()));
```

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 22]
```diff
@@ -0,0 +1,436 @@
...
+namespace dds {
+
+/*
+ * This test checks eprosima::fastdds::statistics::dds::DomainParticipant narrow methods.
+ * 1. Create a eprosima::fastdds::dds::DomainParticipant
+ * 2. Check if the -DFASTDDS_STATISTICS flag has been set and if not check that nullptr is returned even with a valid
+ * DomainParticipant.
+ * 3. Use both narrow methods to obtain the pointer to the children class
+ * 4. Call both narrow methods with an invalid pointer and check that it returns nullptr
+ * 5. Delete DDS entities
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:41]: Check these comments still correspond to the test implementation

File: test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp [hunk 23]
```diff
@@ -0,0 +1,436 @@
...
+ * RETCODE_INCONSISTENT_POLICY.
+ * The case where the create_datawriter fails returning RETCODE_ERROR is not checked because it only passes the error
+ * upstream.
+ * 9. Try to delete the DomainParticipant and check that fails with RETCODE_PRECONDITION_NOT_MET because there are still
+ * statistics DataWriters enabled.
+ * 10. Try to disable an already disabled statistics DataWriter and check that returns RETCODE_ERROR.
+ * 11. Check that if an invalid topic name is provided to the disable_statistics_datawriter method, it returns
+ * RETCODE_BAD_PARAMETER.
+ * 12. Delete DDS entities.
+ * The case where the delete_datawriter fails returning RETCODE_ERROR is not checked because it only passes the error
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp:105]: Check these comments still correspond to the test implementation

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**